mehta-lab/recOrder
# Standalone auto-generated reconstruction GUI

<img src="https://github.com/user-attachments/assets/ce546481-dab7-4bae-a4e2-daa36aeb146e" width="350"/>
<img src="https://github.com/user-attachments/assets/2952c0cf-c1b1-4c5b-8881-541cd4a5a8b0" width="350"/>


<details>
<summary>Quick guide to install the test version:</summary>

Create a new conda environment and activate it
```
conda create -y -n recOrder-test python=3.10
conda activate recOrder-test
```

Install current main branch with the [acq] optional flag
```
git clone https://github.com/mehta-lab/recOrder.git
pip install -e .[acq]
```

run napari to test
`napari`

If you get an error you might need to install napari with its Qt bindings:
`python -m pip install "napari[pyqt6]"`

Make sure napari runs now. If you want to test the current main branch version, you can test that from the plugins menu. Close napari.

We will now install the version for testing from the branch using:
`pip install git+https://github.com/mehta-lab/recOrder.git@pydantic-model-prototype`

<details>
<summary>If you are on a Windows machine, we will need a different submitit package (ignore for other OS)</summary>

```
pip uninstall submitit
pip install git+https://github.com/amitabhverma/submitit.git@windows_fix
```

</details>
</details>

<details>
<summary>Quick guide on Reconstruction:</summary>

1. Select your zarr input storage location.
2. Select which processing mode(s) you want (birefringence, phase, birefringence & phase, fluorescence) and use the Create Model button to add how many you need.
3. For each added model set your parameters and your output storage location.
4. Once ready to process, hit the Build & Run button which will start the Reconstruction.
5. If there are any parameters that fail validation the model with the error will have an alert symbol and in the model GUI the field will be highlighted.
6. If any model fails validation the Reconstruction will not proceed.
7. Once all models are validated the Reconstruction will proceed and the processing updates will pop up in the bottom section. 

Once each process successfully completes that table entry will be purged and only those that might have errored should remain.
</details>


> [!NOTE] 
I would suggest for the first test, run using 1 model and make sure no errors are encountered and the data processes as expected. If you are running on a local machine I also suggest keeping the task manager open to make sure the OS is not running out of resources which might lead to a crash when running multiple processing concurrently.
